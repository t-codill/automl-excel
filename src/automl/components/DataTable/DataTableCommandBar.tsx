import { DefaultButton, IOverflowSetItemProps, OverflowSet, SearchBox } from "office-ui-fabric-react";
import * as React from "react";

export interface IDataTableCommandItem {
    key: string;
    name: string;
    icon?: string;
    onClick(): void;
}

export interface IDataTableCommandBarProps {
    hideFilterBox?: boolean;
    commandItems?: IDataTableCommandItem[];
    overflowItems?: IDataTableCommandItem[];
    onFilterBoxTextChanged?(text: string | undefined): void;
}

export class DataTableCommandBar extends React.Component<IDataTableCommandBarProps>{
    public render(): React.ReactNode {
        let items: IOverflowSetItemProps[] = [];
        let overflowItems: IOverflowSetItemProps[] = [];

        if (!this.props.hideFilterBox) {
            items.push({
                key: "search",
                class: "test",
                onRender: this.renderSearchBox
            });
        }

        if (this.props.commandItems) {
            items = [...items, ...this.props.commandItems];
        }

        if (this.props.overflowItems) {
            overflowItems = [...overflowItems, ...this.props.overflowItems];
        }

        let className = "commands";
        if (!this.props.hideFilterBox) {
            className += " has-filter";
        }

        return <OverflowSet
            className={className}
            items={items}
            overflowItems={overflowItems}
            onRenderItem={this.onRenderItem}
            onRenderOverflowButton={this.onRenderOverflowButton}
        />;
    }

    public readonly onRenderItem = (item: IOverflowSetItemProps) => {
        if (item.onRender) {
            return item.onRender(item);
        }
        return <DefaultButton iconProps={{ iconName: item.icon }} menuProps={item.subMenuProps} text={item.name} onClick={item.onClick} />;
    }

    public readonly onRenderOverflowButton = () => {
        return (
            <DefaultButton menuIconProps={{ iconName: "More" }} />
        );
    }

    private readonly renderSearchBox = () => {
        return <SearchBox placeholder="Search to filter items..."
            className="searchBox"
            onChange={this.props.onFilterBoxTextChanged}
            iconProps={{ iconName: "Filter" }}
            underlined={true}
            autoComplete="off" />;
    }
}
