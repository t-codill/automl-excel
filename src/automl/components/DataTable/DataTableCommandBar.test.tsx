import { shallow } from "enzyme";
import * as React from "react";
import { DataTableCommandBar, IDataTableCommandBarProps } from "./DataTableCommandBar";

describe("DataTableCommandBar", () => {
    describe("Component", () => {
        it("should render", () => {
            const wrapper = shallow(<DataTableCommandBar />);
            expect(wrapper)
                .toMatchSnapshot();
        });
    });
    describe("filterBox", () => {
        it("should not render if hideFilter is set", () => {
            const props: IDataTableCommandBarProps = {
                hideFilterBox: true
            };
            const wrapper = shallow(<DataTableCommandBar {...props} />);
            expect(wrapper)
                .toMatchSnapshot();
        });
    });
    describe("command items", () => {
        it("should render added items", () => {
            const callbackFunc = () => { return; };
            const commandItems = [{ key: "key1", name: "name1", onClick: callbackFunc },
                                  { key: "key2", name: "name2", onClick: callbackFunc }];
            const props: IDataTableCommandBarProps = {
                commandItems
            };
            const wrapper = shallow(<DataTableCommandBar {...props} />);
            expect(wrapper.render())
                .toMatchSnapshot();
        });
        it("should render added items with filter disabled", () => {
            const callbackFunc = () => { return; };
            const commandItems = [{ key: "key1", name: "name1", onClick: callbackFunc },
                                  { key: "key2", name: "name2", onClick: callbackFunc }];
            const props: IDataTableCommandBarProps = {
                hideFilterBox: true,
                commandItems
            };
            const wrapper = shallow(<DataTableCommandBar {...props} />);
            expect(wrapper.render())
                .toMatchSnapshot();
        });
        it("should render overflow items", () => {
            const callbackFunc = () => { return; };

            const commandItems = [{ key: "key1", name: "name1", onClick: callbackFunc },
                                  { key: "key2", name: "name2", onClick: callbackFunc }];

            const overflowItems = [{ key: "key3", name: "name3", onClick: callbackFunc },
                                   { key: "key4", name: "name4", onClick: callbackFunc }];

            const props: IDataTableCommandBarProps = {
                commandItems,
                overflowItems
            };
            const wrapper = shallow(<DataTableCommandBar {...props} />);
            expect(wrapper.render())
                .toMatchSnapshot();
        });
        it("should render overflow items with filter disabled", () => {
            const callbackFunc = () => { return; };

            const commandItems = [{ key: "key1", name: "name1", onClick: callbackFunc },
                                  { key: "key2", name: "name2", onClick: callbackFunc }];

            const overflowItems = [{ key: "key3", name: "name3", onClick: callbackFunc },
                                   { key: "key4", name: "name4", onClick: callbackFunc }];

            const props: IDataTableCommandBarProps = {
                hideFilterBox: true,
                commandItems,
                overflowItems
            };
            const wrapper = shallow(<DataTableCommandBar {...props} />);
            expect(wrapper.render())
                .toMatchSnapshot();
        });
    });
});
