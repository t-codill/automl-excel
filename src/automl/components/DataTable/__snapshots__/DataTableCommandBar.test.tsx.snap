// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTableCommandBar Component should render 1`] = `
<StyledOverflowSetBase
  className="commands has-filter"
  items={
    Array [
      Object {
        "class": "test",
        "key": "search",
        "onRender": [Function],
      },
    ]
  }
  onRenderItem={[Function]}
  onRenderOverflowButton={[Function]}
  overflowItems={Array []}
/>
`;

exports[`DataTableCommandBar command items should render added items 1`] = `
<div
  class="ms-FocusZone css-50 ms-OverflowSet commands has-filter root-47"
  data-focuszone-id="FocusZone1"
  role="menubar"
>
  <div
    class="ms-OverflowSet-item item-48"
  >
    <div
      class="ms-SearchBox is-underlined searchBox root-51"
    >
      <div
        aria-hidden="true"
        class="ms-SearchBox-iconContainer iconContainer-52"
      >
        <i
          class="ms-SearchBox-icon icon-57"
          data-icon-name="Filter"
          role="presentation"
        >
          
        </i>
      </div>
      <input
        aria-label="Search to filter items..."
        autocomplete="off"
        class="ms-SearchBox-field field-55"
        id="SearchBox2"
        placeholder="Search to filter items..."
        value=""
      />
    </div>
  </div>
  <div
    class="ms-OverflowSet-item item-48"
  >
    <button
      class="ms-Button ms-Button--default root-58"
      data-is-focusable="true"
      type="button"
    >
      <div
        class="ms-Button-flexContainer flexContainer-59"
      >
        <div
          class="ms-Button-textContainer textContainer-60"
        >
          <div
            class="ms-Button-label label-62"
            id="id__3"
          >
            name1
          </div>
        </div>
      </div>
    </button>
  </div>
  <div
    class="ms-OverflowSet-item item-48"
  >
    <button
      class="ms-Button ms-Button--default root-58"
      data-is-focusable="true"
      type="button"
    >
      <div
        class="ms-Button-flexContainer flexContainer-59"
      >
        <div
          class="ms-Button-textContainer textContainer-60"
        >
          <div
            class="ms-Button-label label-62"
            id="id__6"
          >
            name2
          </div>
        </div>
      </div>
    </button>
  </div>
</div>
`;

exports[`DataTableCommandBar command items should render added items with filter disabled 1`] = `
<div
  class="ms-FocusZone css-50 ms-OverflowSet commands root-47"
  data-focuszone-id="FocusZone9"
  role="menubar"
>
  <div
    class="ms-OverflowSet-item item-48"
  >
    <button
      class="ms-Button ms-Button--default root-58"
      data-is-focusable="true"
      type="button"
    >
      <div
        class="ms-Button-flexContainer flexContainer-59"
      >
        <div
          class="ms-Button-textContainer textContainer-60"
        >
          <div
            class="ms-Button-label label-62"
            id="id__10"
          >
            name1
          </div>
        </div>
      </div>
    </button>
  </div>
  <div
    class="ms-OverflowSet-item item-48"
  >
    <button
      class="ms-Button ms-Button--default root-58"
      data-is-focusable="true"
      type="button"
    >
      <div
        class="ms-Button-flexContainer flexContainer-59"
      >
        <div
          class="ms-Button-textContainer textContainer-60"
        >
          <div
            class="ms-Button-label label-62"
            id="id__13"
          >
            name2
          </div>
        </div>
      </div>
    </button>
  </div>
</div>
`;

exports[`DataTableCommandBar command items should render overflow items 1`] = `
<div
  class="ms-FocusZone css-50 ms-OverflowSet commands has-filter root-47"
  data-focuszone-id="FocusZone16"
  role="menubar"
>
  <div
    class="ms-OverflowSet-item item-48"
  >
    <div
      class="ms-SearchBox is-underlined searchBox root-51"
    >
      <div
        aria-hidden="true"
        class="ms-SearchBox-iconContainer iconContainer-52"
      >
        <i
          class="ms-SearchBox-icon icon-57"
          data-icon-name="Filter"
          role="presentation"
        >
          
        </i>
      </div>
      <input
        aria-label="Search to filter items..."
        autocomplete="off"
        class="ms-SearchBox-field field-55"
        id="SearchBox17"
        placeholder="Search to filter items..."
        value=""
      />
    </div>
  </div>
  <div
    class="ms-OverflowSet-item item-48"
  >
    <button
      class="ms-Button ms-Button--default root-58"
      data-is-focusable="true"
      type="button"
    >
      <div
        class="ms-Button-flexContainer flexContainer-59"
      >
        <div
          class="ms-Button-textContainer textContainer-60"
        >
          <div
            class="ms-Button-label label-62"
            id="id__18"
          >
            name1
          </div>
        </div>
      </div>
    </button>
  </div>
  <div
    class="ms-OverflowSet-item item-48"
  >
    <button
      class="ms-Button ms-Button--default root-58"
      data-is-focusable="true"
      type="button"
    >
      <div
        class="ms-Button-flexContainer flexContainer-59"
      >
        <div
          class="ms-Button-textContainer textContainer-60"
        >
          <div
            class="ms-Button-label label-62"
            id="id__21"
          >
            name2
          </div>
        </div>
      </div>
    </button>
  </div>
  <div
    class="ms-OverflowSet-overflowButton overflowButton-49"
  >
    <button
      class="ms-Button ms-Button--default root-58"
      data-is-focusable="true"
      type="button"
    >
      <div
        class="ms-Button-flexContainer flexContainer-59"
      >
        <i
          class="ms-Button-menuIcon menuIcon-66"
          data-icon-name="More"
          role="presentation"
        >
          
        </i>
      </div>
    </button>
  </div>
</div>
`;

exports[`DataTableCommandBar command items should render overflow items with filter disabled 1`] = `
<div
  class="ms-FocusZone css-50 ms-OverflowSet commands root-47"
  data-focuszone-id="FocusZone27"
  role="menubar"
>
  <div
    class="ms-OverflowSet-item item-48"
  >
    <button
      class="ms-Button ms-Button--default root-58"
      data-is-focusable="true"
      type="button"
    >
      <div
        class="ms-Button-flexContainer flexContainer-59"
      >
        <div
          class="ms-Button-textContainer textContainer-60"
        >
          <div
            class="ms-Button-label label-62"
            id="id__28"
          >
            name1
          </div>
        </div>
      </div>
    </button>
  </div>
  <div
    class="ms-OverflowSet-item item-48"
  >
    <button
      class="ms-Button ms-Button--default root-58"
      data-is-focusable="true"
      type="button"
    >
      <div
        class="ms-Button-flexContainer flexContainer-59"
      >
        <div
          class="ms-Button-textContainer textContainer-60"
        >
          <div
            class="ms-Button-label label-62"
            id="id__31"
          >
            name2
          </div>
        </div>
      </div>
    </button>
  </div>
  <div
    class="ms-OverflowSet-overflowButton overflowButton-49"
  >
    <button
      class="ms-Button ms-Button--default root-58"
      data-is-focusable="true"
      type="button"
    >
      <div
        class="ms-Button-flexContainer flexContainer-59"
      >
        <i
          class="ms-Button-menuIcon menuIcon-66"
          data-icon-name="More"
          role="presentation"
        >
          
        </i>
      </div>
    </button>
  </div>
</div>
`;

exports[`DataTableCommandBar filterBox should not render if hideFilter is set 1`] = `
<StyledOverflowSetBase
  className="commands"
  items={Array []}
  onRenderItem={[Function]}
  onRenderOverflowButton={[Function]}
  overflowItems={Array []}
/>
`;
